server/src/main/java/io/crate/expression/scalar/SubscriptFunction.java

```diff
@@ -192,13 +202,18 @@ static Object lookupByNumericIndex(Object base, Object index) {
         }
     }
 
-    static Object lookupByName(Object base, Object name) {
+    static Object lookupByName(Object base, Object name, Boolean errorOnUnknownObjectKey) {
         if (!(base instanceof Map)) {
             throw new IllegalArgumentException("Base argument to subscript must be an object, not " + base);
         }
         Map<?, ?> map = (Map<?, ?>) base;
-        if (!map.containsKey(name)) {
-            throw new IllegalArgumentException("The object `" + base + "` does not contain the key `" + name + "`");
+        if (errorOnUnknownObjectKey && !map.containsKey(name)) {
+            throw new ColumnUnknownException(base + "['" + name + "']", null) {
+                @Override
+                public Iterable<RelationName> getTableIdents() {
+                    throw new RelationUnknown("unknown relation name");
```

Comment:
I think this should return an empty iterable instead. There is no relation involved, so it doesn't have to return any.